lib/std/target.zig

```diff
@@ -550,6 +550,103 @@ pub const Target = struct {
         }
     };
 
+    /// processor specific ABI
+    pub const TargetAbi = enum {
+        //TODO add ARM, Mips, and PowerPC
+        ilp32,
+        ilp32d,
+        ilp32e,
+        ilp32f,
+        lp64,
+        lp64d,
+        lp64f,
+
+        const Riscv32ABI = struct {
+            fn default(features: Cpu.Feature.Set) TargetAbi {
+                if (riscv.featureSetHas(features, .d)) {
+                    return .ilp32d;
+                } else if (riscv.featureSetHas(features, .e)) {
+                    return .ilp32e;
+                } else {
+                    return .ilp32;
+                }
+            }
+            fn validate(target_abi: TargetAbi, features: Cpu.Feature.Set) ParseError!void {
+                const has_e = riscv.featureSetHas(features, .e);
+                const has_f = riscv.featureSetHas(features, .f);
+                const has_d = riscv.featureSetHas(features, .d);
+                return switch (target_abi) {
+                    .ilp32e => if (has_e) void{} else error.FeatureAbiMismatch,
```

Comment:
Huh, is `void{}` even valid Zig code? `void` is a type and `{}` an empty block, it's not really meaningful.

Reply:
It compiles and works. I assume it works like regular type instantiation. Though it's not used often, from grep I see it's only being used in the tests in test/ .

In this case it's just being used in place of a missing error. I guess I could probably use an ?error as a return type instead. Or maybe there's a better way?

Reply:
Colour me surprised, that seems an unwanted construct that's accepted by the compiler, if you try replacing `void` with some other type you'll see it doesn't really work.

You can simply replace `void{}` with `{}`.

Reply:
Ah, I had been thinking like struct initialization.
Like
```zig
const Nothing = struct {};
var a = Nothing{};
```

It's interesting to find out that `{}` has the type of `void`. I'll change these `void{}` to `{}`.